packages/integration-aws-node/src/DefaultAwsCredentialsManager.test.ts

```diff
@@ -127,12 +135,22 @@ describe('DefaultAwsCredentialsManager', () => {
         },
       });
 
+    const testDate = new Date('2022-01-10');
+
     process.env.AWS_ACCESS_KEY_ID = 'ACCESS_KEY_ID_10';
     process.env.AWS_SECRET_ACCESS_KEY = 'SECRET_ACCESS_KEY_10';
     process.env.AWS_SESSION_TOKEN = 'SESSION_TOKEN_10';
-    process.env.AWS_CREDENTIAL_EXPIRATION = new Date(
-      '2022-01-10',
-    ).toISOString();
+    process.env.AWS_CREDENTIAL_EXPIRATION = testDate.toISOString();
+
+    // Return creds from env
+    (fromNodeProviderChain as jest.Mock).mockReturnValue(() =>
+      Promise.resolve({
+        accessKeyId: process.env.AWS_ACCESS_KEY_ID,
```

Comment:
Hm. This bypasses the original test's intent to exercise the _actual_ `fromNodeProviderChain` implementation to see that we get the env vars passed through the way that we expect.

You could maybe use spyOn on just this function?

```ts
import * as foo from 'module';
const spy = jest.spyOn(foo, 'functionName');
```

or if that fails, use the mocking method that you already have, but do essentially `jest.fn().mockImplementation(jest.requireActual('foo').bar)` or so.

Then the real impl should be called, while you can still observe what it was called with.

Reply:
Okay, I've updated the mocked code to your second suggestion as `spy` wouldn't work.

```js
    (fromNodeProviderChain as jest.Mock).mockImplementation(
      jest.requireActual('@aws-sdk/credential-providers').fromNodeProviderChain,
    );
```